### DISCUSSION TITLE
Question: Virtual Blynk Zones name

### DISCUSSION THREAD

**Discussion by l33d** [taligentx/dscKeybusInterface#74] [dscKeybusInterface Discussion]
Hi @taligentx 

I have a question about zones info. How can i rename zones info on lcd? Something like from this: zone 2 Open to Kitchen Open? Only in blynk or mus change in sketch? How can i do this?
Next question. How can i made a translate the information on the lcd? Can i do this?

Cheers

**Comment by taligentx** [comment]
Hi @l33d - the Blynk LCD only shows "Zones open" and doesn't specify a zone number, the zone LEDs handle that part.  You can set custom names if you change the sketch in the `setStatus()` function.

You would need to change:
`case 0x03: lcd.print(0,1, "                "); lcd.print(0,1, "Zones open"); break;`

To something like:
```
    case 0x03: {
        if (bitRead(dsc.openZones[0], 0)) {
            lcd.print(0,1, "                "); lcd.print(0,1, "Zone 1 open");
        }
        else if (bitRead(dsc.openZones[0], 1)) {
            lcd.print(0,1, "                "); lcd.print(0,1, "Zone 2 open");
        }
        else {
            lcd.print(0,1, "                "); lcd.print(0,1, "Zones open");
        }
        break;
    }
```

This is untested but that's the idea - for each zone that you would like a custom label, have the sketch check if the zone is open and then print a custom label.  See the sketch section that updates the zone LEDs for how to check if a particular zone is open.

The `setStatus()` function is also where you can translate the Blynk text - it would be nice to have a separate file with different translations, added to the todo list.

**Comment by l33d** [comment]
Hi,

Big thanks, i try in this week ;)

If i understand i must change this 

> case 0x03: {
>         if (bitRead(dsc.openZones[0], 0)) {
>             lcd.print(0,1, "                "); lcd.print(0,1, "Zone 1 open");
>         }
>         else if (bitRead(dsc.openZones[0], 1)) {
>             lcd.print(0,1, "                "); lcd.print(0,1, "Zone 2 open");
>         }
>         else {
>             lcd.print(0,1, "                "); lcd.print(0,1, "Zones open");
>         }
>         break;
>     }

to this 

> case 0x03: {
>         if (bitRead(dsc.openZones[0], 0)) {
>             lcd.print(0,1, "                "); lcd.print(0,1, "Strefa 1 otwarta");
>         }
>         else if (bitRead(dsc.openZones[0], 1)) {
>             lcd.print(0,1, "                "); lcd.print(0,1, "Strefa 2 otwarta");
>         }
>         else {
>             lcd.print(0,1, "                "); lcd.print(0,1, "Strefa otwarta");
>         }
>         break;
>     }
> 

and 
translate others function like armed and disarmed, etc. like this

> void setStatus(byte partition) {
>   static byte lastStatus[8];
>   if (!partitionChanged && dsc.status[partition] == lastStatus[partition]) return;
>   lastStatus[partition] = dsc.status[partition];
> 
>   switch (dsc.status[partition]) {
>     case 0x01: lcd.print(0,1, "                "); lcd.print(0,1, "Ready > Gotowy"); break;
>     case 0x02: lcd.print(0,1, "                "); lcd.print(0,1, "Stay zones open > ..."); break;
>     case 0x03: lcd.print(0,1, "                "); lcd.print(0,1, "Zones open"); break;
>     case 0x04: lcd.print(0,1, "                "); lcd.print(0,1, "Armed stay"); break;
>     case 0x05: lcd.print(0,1, "                "); lcd.print(0,1, "Armed away"); break;
>     case 0x07: lcd.print(0,1, "                "); lcd.print(0,1, "Failed to arm"); break;
>     case 0x08: lcd.print(0,1, "                "); lcd.print(0,1, "Exit delay >...""); break;
>     case 0x09: lcd.print(0,1, "                "); lcd.print(0,1, "No entry delay >...""); break;
>     case 0x0B: lcd.print(0,1, "                "); lcd.print(0,1, "Quick exit >...""); break;
>     case 0x0C: lcd.print(0,1, "                "); lcd.print(0,1, "Entry delay >...""); break;
>     case 0x0D: lcd.print(0,1, "                "); lcd.print(0,1, "Alarm memory > Pamięć zdarzeń"); break;
>     case 0x10: lcd.print(0,1, "                "); lcd.print(0,1, "Keypad lockout >..."); break;


?

**Comment by taligentx** [comment]
That should work, though as I mentioned I haven't actually tested the code.  You'll need to create a new section for each zone in that `if... else if... else` section for open zones.  You're right about the translation for the `setStatus()` function, for the Blynk sketch all of the terms are in the sketch.

**Comment by l33d** [comment]
Ok, thanks i try and test.

**Comment by l33d** [comment]
Hi,
I have only translated statuses in 95%. Unfortunately, today I tried to change the names of the zones as shown in the example, but it does not work, or I do not understand something or it will not work like this :(
First metod i try like this:

> case 0x50: {
>     if (bitRead(dsc.openZones[0], 0)) {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zone 1 open");
>     }
>     else if (bitRead(dsc.openZones[0], 1)) {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zone 2 open");
>     }
>     else {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zones open");
>     }
>     
>      break; }
>      
>       case 0x51: {
>     if (bitRead(dsc.openZones[0], 2)) {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zone 3 open");
>     }
>     else if (bitRead(dsc.openZones[0], 3)) {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zone 4 open");
>     }
>     else {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zones open");
>     }
>      
>     
>        break; }
>   
> 
>      case 0x52: {
>     if (bitRead(dsc.openZones[0], 4)) {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zone 5 open");
>     }
>     else if (bitRead(dsc.openZones[0], 5)) {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zone 6 open");
>     }
>     else {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zones open");
>     }
>     
>      break; }
> 
>       case 0x53: {
>     if (bitRead(dsc.openZones[0], 6)) {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zone 7 open");
>     }
>     else if (bitRead(dsc.openZones[0], 7)) {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zone 8 open");
>     }
>     else {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zones open");
>     }
>     
>      break; }

and second 

> case 0x50: {
>     if (bitRead(dsc.openZones[0], 0)) {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zone 1 open");
>     }
>     else if (bitRead(dsc.openZones[0], 1)) {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zone 2 open");
>     }
>     else {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zones open");
>     }
>     
>      break; }
>      
>       case 0x51: {
>     if (bitRead(dsc.openZones[0], 2)) {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zone 3 open");
>     }
>     else if (bitRead(dsc.openZones[0], 3)) {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zone 4 open");
>     }
>     else {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zones open");
>     }
>      
>      
>        break; }
>   
> 
>      case 0x52: {
>     if (bitRead(dsc.openZones[0], 4)) {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zone 5 open");
>     }
>     else if (bitRead(dsc.openZones[0], 5)) {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zone 6 open");
>     }
>     else {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zones open");
>     }
>     
>      
>        break; }
> 
>       case 0x53: {
>     if (bitRead(dsc.openZones[0], 6)) {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zone 7 open");
>     }
>     else if (bitRead(dsc.openZones[0], 7)) {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zone 8 open");
>     }
>     else {
>     lcd.print(0,1, " "); lcd.print(0,1, "Zones open");
>     }
>     
>     
>      break; }



I wanted to add my translation of the pushbullet and virtualblynk sketch, but I do not know how to do it on github. I tried, but something did not work out and did not add it.

Last question about virtual blynk. For example, I enter the alarm memory * 3, how can I see what happened? On lcd5500Z I can see text and what can be done. In an ordinary LED manipulator, leds are lit, which inform me what happened, but how can see information about that in virtual blynk?

Cheers,
